miio/tests/test_device.py

```diff
@@ -17,6 +17,17 @@ def test_get_properties_splitting(mocker, max_properties):
     if max_properties is None:
         max_properties = len(properties)
     assert send.call_count == math.ceil(len(properties) / max_properties)
+    assert 5 == d._protocol._timeout
+
+
+def test_timeout_retry(mocker):
+    send = mocker.patch("miio.miioprotocol.MiIOProtocol.send")
+    d = Device("127.0.0.1", "68ffffffffffffffffffffffffffffff", timeout=4)
+    assert 4 == d._protocol._timeout
+    d.send("fake_command", [], 1)
+    send.assert_called_with("fake_command", [], 1, extra_parameters=None)
```

Comment:
I'm not sure if `assert_called_with` also works with kwargs, but if it does, prefer the kwarg format for calls & checks.

It would also be great to have separate tests for setting the timeout and the retry counts in a subclass, you could add a `Device`-derived test class, with some custom timeout & retry count values, to this file and use that for these tests.

Reply:
it does, but `send` is not called with kwargs only (see: https://github.com/rytilahti/python-miio/blob/master/miio/device.py#L147)